examples/plot_structure_basic/plot_inferred_structure.py

```diff
@@ -0,0 +1,223 @@
+                m = np.arange(j, j + 1 + 0.1, 0.1)
+            else:
+                m = np.arange(j, j + 1, 0.1)
+            smoothed_X[i].append([f_x(np.arange(m.min(), m.max(), 0.1)),
+                                  f_y(np.arange(m.min(), m.max(), 0.1)),
+                                  f_z(np.arange(m.min(), m.max(), 0.1))])
+        
+        begin = end
+        
+    return smoothed_X
+
+
+
+##############################################################################
+# Define the function to scatter the beads.
+def scatterBeads(x, y, z, bead_size, name, curr_cmap, indices, ax):
+    last_idx = len(x) - 1
+    ax.scatter(x[0], y[0], z[0], s=bead_size, color=curr_cmap(indices[0]),
+               label=name + ' start')
+    ax.scatter(x[1:last_idx], y[1:last_idx], z[1:last_idx], s=bead_size,
+               cmap=curr_cmap, c=indices[1:last_idx])
+    ax.scatter(x[last_idx], y[last_idx], z[last_idx], s=bead_size,
+               color=curr_cmap(indices[last_idx]), label=name + ' end')
+
+
+
+##############################################################################
+# Define the function to lighten or darken our color map.
+def cmap_map(function, cmap):
+    # Function taken from:
+    # https://scipy-cookbook.readthedocs.io/items/Matplotlib_ColormapTransformations.html
+    
+    """ Applies function (which should operate on vectors of shape 3:
+        [r, g, b]), on colormap cmap. This routine will break any discontinuous
+        points in a colormap.
+    """
+    cdict = cmap._segmentdata
+    step_dict = {}
+    # Firt get the list of points where the segments start or end
+    for key in ('red', 'green', 'blue'):
+        step_dict[key] = list(map(lambda x: x[0], cdict[key]))
+    step_list = sum(step_dict.values(), [])
+    step_list = np.array(list(set(step_list)))
+    # Then compute the LUT, and apply the function to the LUT
+    reduced_cmap = lambda step : np.array(cmap(step)[0:3])
+    old_LUT = np.array(list(map(reduced_cmap, step_list)))
+    new_LUT = np.array(list(map(function, old_LUT)))
+    # Now try to make a minimal segment definition of the new LUT
+    cdict = {}
+    for i, key in enumerate(['red','green','blue']):
+        this_cdict = {}
+        for j, step in enumerate(step_list):
+            if step in step_dict[key]:
+                this_cdict[step] = new_LUT[j, i]
+            elif new_LUT[j,i] != old_LUT[j, i]:
+                this_cdict[step] = new_LUT[j, i]
+        colorvector = list(map(lambda x: x + (x[1], ), this_cdict.items()))
+        colorvector.sort()
+        cdict[key] = colorvector
+
+    return matplotlib.colors.LinearSegmentedColormap('colormap',cdict,1024)
+
+
+
+##############################################################################
+# Turn the dataframe into an array of coordinates for the two chromosomes,
+# both of which have length 60 (the array is length 120). Then, get the
+# coordinates for each chromosome.
+coordArr = np.array(df)
```

Comment:
it's better to use df.values here. `coordArr` doesn't follow python conventions for naming: it should be all lower case with underscores (ie, `coord` or `coordinates` would be fine, or `coord_arr` if you want to underline it's an array).

Reply:
Renamed to coordinates.